dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1002 @@
+
+  def _SearchForDeletedResource(
+      self, resource: Resource,
+      start_timestamp: datetime) -> Optional[Resource]:
+    """Searches for deleted resource in GCP Logs.
+
+    Args:
+      resource: resource to search for.
+      start_timestamp: the initial point of time to start the search
+
+    Return:
+      Found resource or None
+    """
+    if not resource or not start_timestamp:
+      return None
+
+    if resource.project_id != self.project_id:
+      return None
+
+    while start_timestamp > (datetime.now(timezone.utc) - timedelta(days=400)):
+
+      end_timestamp = start_timestamp + timedelta(minutes=20)
+      start_timestamp = start_timestamp - timedelta(days=30)
+
+      log_messages = self._GetLogMessages(resource.project_id, start_timestamp,
+                                          end_timestamp, resource.id)
+      self._ParseLogMessages(log_messages)
+
+      matched_resource = self._FindResource(resource.name, resource.type)
+      if matched_resource:
+        if matched_resource.deletion_timestamp and matched_resource.creation_timestamp:  # pylint: disable=line-too-long
+          return matched_resource
+
+    return None
+
+  def _RetrieveListOfDisks(self, project_id: str) -> Dict[str, Resource]:
+    """Retrieves list of disks in a project.
+
+    Args:
+      project_id: Project Id to retrieve the list of disks for
+
+    Returns:
+      Dict of disks
+    """
+    result: Dict[str, Resource] = {}
+
+    # Using beta version of the API because v1 did not have important
+    # information when creating this script
+    compute_api_client = CreateService('compute', 'beta')
+
+    request = compute_api_client.disks().aggregatedList(project=project_id)  # pylint: disable=no-member
+
+    while request is not None:
+      response = request.execute()
+
+      for zone in response.get('items',{}).values():
+
+        for disk in zone.get('disks', {}):
+          resource = Resource()
+          resource.id = disk.get('id')
+          resource.resource_name = disk.get('selfLink')
+          resource.creation_timestamp = disk.get('creationTimestamp')
+          if disk.get('sourceDisk'):
+            resource.parent = Resource()
+            resource.parent.resource_name = disk.get('sourceDisk')
+            resource.parent.id = disk.get('sourceDiskId')
+          elif disk.get('sourceSnapshot'):
+            resource.parent = Resource()
+            resource.parent.resource_name = disk.get('sourceSnapshot')
+            resource.parent.id = disk.get('sourceSnapshotId')
+          elif disk.get('sourceImage'):
+            resource.parent = Resource()
+            resource.parent.resource_name = disk.get('sourceImage')
+            resource.parent.id = disk.get('sourceImageId')
+
+          result[resource.id] = resource
+
+      # pylint: disable=no-member
+      request = compute_api_client.disks().aggregatedList_next(
+          previous_request=request, previous_response=response)
+
```

Comment:
calling `ListDisks` https://github.com/google/cloud-forensics-utils/blob/719093b4a229e5e97c30d93faabb1ccf3b6ee422/libcloudforensics/providers/gcp/internal/compute.py#L242 
will do all that and it will return a dict of Dict[disk_name[str]: disk[GoogleComputeDisk]]
you can then do disk.GetOperation() and get the full api object of that disk or you can call .keys() on the dict and get names of all disks